presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -561,13 +563,18 @@ private SliceReadFunction arrayAsJsonReadFunction(ConnectorSession session, Colu
             // read array into a block
             Block block = ((BlockReadFunction) readFunction).readBlock(resultSet, columnIndex);
 
-            // cast block to JSON slice
+            // convert block to JSON slice
+            Type blockType = ((ArrayType) type).getElementType();
+            List<Object> values = new ArrayList<>();
+            for (int position = 0; position < block.getPositionCount(); position++) {
+                values.add(blockType.getObjectValue(session, block, position));
+            }
+
             try {
-                return (Slice) typeManager.getCoercion(type, jsonType)
-                        .invokeExact(session, block);
+                return Slices.wrappedBuffer(SORTED_MAPPER.writeValueAsBytes(values));
             }
-            catch (Throwable throwable) {
-                throw new PrestoException(JDBC_ERROR, "Cast to JSON failed: " + throwable.getMessage(), throwable);
```

Comment:
i think it's useful to incluide `throwable.getMessage()` so that people don't need to `--debug` to know _something_. "Cast to JSON failed" alone is not really informative / actionable.

(i know you prefer not to. I am ok either way)

Reply:
It depends on the context. Is the message meaningful to end users? In this case, I don't think it is, since we don't expect this to fail. If it does, that's a bug somewhere, and we probably need the stack trace and other information like the input data.

If this was an error parsing user supplied JSON, then providing the message makes sense.